### PR TITLE
Replace cli with cli-ng

### PR DESCRIPTION
**Pull request by ernestask**
Related: https://github.com/abrt/abrt/issues/1394

Signed-off-by: Ernestas Kulik <ekulik@redhat.com>

### CODE REVIEW

File: abrt.spec.in [hunk 1]
```diff
@@ -361,45 +350,12 @@ Requires: python3-argh
...
-Requires: libreport-plugin-bugzilla >= %{libreport_ver}
-%endif
-Requires: libreport-plugin-logger >= %{libreport_ver}
-Requires: libreport-plugin-ureport >= %{libreport_ver}
-%if 0%{?fedora}
-Requires: libreport-fedora >= %{libreport_ver}
-%endif
-%endif
+Obsoletes: %{name}-cli-ng <= 2.12.0
+Obsoletes: %{name}-tui <= 2.12.0
```
Comment (mkutlak) [inline abrt.spec.in]: You should add `Provides:` for these two packages. Otherwise you will have trouble upgrading.

Comment (ernestask) [reply abrt.spec.in]: I’m not sure that’s right. Have you tried the upgrade path? Works fine for me.

Comment (ernestask) [reply abrt.spec.in]: Should be fixederino nowerino.

**Comment by mkutlak** [comment]
This will break the integration tests.

1) `abrt` returns `No problems` if there are no problems but `abrt-cli` returns nothing.
https://github.com/abrt/abrt/blob/b12aef3da080b284cdb33f8aa1288e468346b21c/tests/runtests/aux/lib.sh#L4

```
$ abrt-cli ls

$ abrt ls
No problems
```

2) `abrt` doesn't have functionality to match dump dirs.
``` bash
rlRun "abrt-cli rm $crash_PATH" 0 "Remove crash dir"
```

3) Differences in outputs and commands needed:
https://github.com/abrt/abrt/blob/b12aef3da080b284cdb33f8aa1288e468346b21c/tests/runtests/aux/lib.sh#L40
``` bash
$ abrt ls
Id           b42c8ec  
Component    will-crash  
Count        1  
Time         2019-07-08 15:21:28  
User id      1000 (mkutlak)  
Reported to    
ABRT Server  https://retrace.fedoraproject.org/faf/reports/bthash/409cbe6c252ec8b2e12b29150acdd179be4bb66f

$ abrt ls --pretty full
Id            b42c8ec  
Component     will-crash  
Count         1  
Time          2019-07-08 15:21:28  
Command line  will_segfault  
Package       will-crash-0.10-10.fc30  
User id       1000 (mkutlak)  
Path          /var/spool/abrt/ccpp-2019-07-08-15:21:28-32352  
Reported to     
ABRT Server   https://retrace.fedoraproject.org/faf/reports/bthash/409cbe6c252ec8b2e12b29150acdd179be4bb66f

$ abrt-cli ls
id b42c8ec9230e80896311c44755219750cfd48ea7
reason:         will_segfault killed by SIGSEGV
time:           Mon 08 Jul 2019 03:21:28 PM CEST
cmdline:        will_segfault
package:        will-crash-0.10-10.fc30
uid:            1000 (mkutlak)
count:          1
Directory:      /var/spool/abrt/ccpp-2019-07-08-15:21:28-32352
Reported:       https://retrace.fedoraproject.org/faf/reports/bthash/409cbe6c252ec8b2e12b29150acdd179be4bb66f
```
ie.: `Directory` vs `Path`

**Comment by ernestask** [comment]
> `abrt` returns `No problems` if there are no problems but `abrt-cli` returns nothing.

Myeah, I was going to address it later. For the test, using that command is a bit lame. `status --bare` would be better IMO.

> `abrt` doesn't have functionality to match dump dirs.

Indeed. The “get crash path” logic in the tests is incredibly flaky, so maybe it would also be possible to come up with something better while I’m at it.

> Differences in outputs and commands needed

Sure.

File: abrt.spec.in [hunk 2]
```diff
@@ -357,45 +346,15 @@ Requires: python3-argh
...
-Requires: libreport-fedora >= %{libreport_ver}
-%endif
-%endif
+Provides:  %{name}-tui = %{version}-%{release}
+Obsoletes: %{name}-tui <= 2.12.1
 
 %description cli
-Virtual package to install all necessary packages for usage from command line
-environment.
+New generation command line interface for ABRT
```
Comment (xsuchy) [inline abrt.spec.in]: Can we get rid of "new generation"? It is, in fact, already old :)

Comment (ernestask) [reply abrt.spec.in]: “Old generation” it is, gotcha.

**Comment by ernestask** [comment]
@trapas, retest this please

**Comment by ernestask** [comment]
To be fair, this should be ready for review now. One stubborn test was failing, but I’m confident that it’s fixed (not that it was anything other than the test being broken).

File: tests/runtests/bz652338-removed-proc-PID/runtest.sh [hunk 3]
```diff
@@ -77,11 +74,11 @@ rlJournalStart
...
         rlAssertGrep "Skipping core dump" var-log-messages
 
-        rlRun "$CLI_LIST | grep package | grep abrt" 1 "abrtd did not crashed"
+        rlRun "abrt list --fmt={package} | grep abrt" 1 "abrtd did not crashed"
 
-        dirs="$($CLI_LIST | grep Directory | awk '{ print $2 }')"
+        dirs="$(abrt list --fmt={path})"
         for dir in $dirs; do
-            rlRun "$CLI_RM $dir" 0 "Dump dir removed ($dir)"
+            rlRun "abrt remove $dir" 0 "Dump dir removed ($dir)"
```
Comment (mkutlak) [inline tests/runtests/bz652338-removed-proc-PID/runtest.sh]: Does this work without `-f`?

Comment (ernestask) [reply tests/runtests/bz652338-removed-proc-PID/runtest.sh]: True, it doesn’t.

Comment (ernestask) [reply tests/runtests/bz652338-removed-proc-PID/runtest.sh]: Wait, it does! I got it mixed up with `abrt remove` with no argument.

Comment (mkutlak) [reply tests/runtests/bz652338-removed-proc-PID/runtest.sh]: Yes, it works.

Comment (ernestask) [reply tests/runtests/bz652338-removed-proc-PID/runtest.sh]: Agree to disagree.

File: tests/runtests/cli-list-suggest-autoreporting/PURPOSE [hunk 4]
```diff
@@ -1,3 +0,0 @@
-PURPOSE of cli-list-suggests-autoreporting
```
Comment (mkutlak) [inline tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: It would be nice to have this in the new cli as well.

Comment (ernestask) [reply tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: It was more or less deliberate, because I didn’t see any value in having that.

Comment (ernestask) [reply tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: Do you remember why it was added to begin with?

Comment (ernestask) [reply tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: Right, so I found https://bugzilla.redhat.com/show_bug.cgi?id=1044424, which doesn’t seem to base the assumption on anything.

Comment (ernestask) [reply tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: I assume that maybe this is targeting anyone who doesn’t use GNOME or any other DE with a first-time setup thing that allows enabling automatic report?

File: abrt.spec.in [hunk 5]
```diff
@@ -357,45 +346,15 @@ Requires: python3-argh
...
-Requires: libreport-plugin-bugzilla >= %{libreport_ver}
-%endif
-Requires: libreport-plugin-logger >= %{libreport_ver}
-Requires: libreport-plugin-ureport >= %{libreport_ver}
-%if 0%{?fedora}
-Requires: libreport-fedora >= %{libreport_ver}
-%endif
-%endif
+Provides:  %{name}-tui = %{version}-%{release}
+Obsoletes: %{name}-tui <= 2.12.1
```
Comment (ernestask) [inline abrt.spec.in]: So this came up yesterday - is the “provides” line correct?

I would say that I should keep the subpackage as is for the purposes of this PR, but the idea behind that was that we shouldn’t offer non-ABRT interfaces in the ABRT package.

File: tests/runtests/cli-list-suggest-autoreporting/PURPOSE (hunk 4 above)
Comment (ernestask) [reply tests/runtests/cli-list-suggest-autoreporting/PURPOSE]: Would moving this nagging to the shell notification script make more sense? I.e. “ABRT has detected X new problems; also, automatic reporting is disabled”.

**Comment by ernestask** [comment]
abrt-cli is back now.

File: src/cli/abrtcli/cli.py [hunk 6]
```diff
@@ -173,10 +173,17 @@ def list_problems(args):
     else:
         print(_('No problems'))
 
+    settings = problem.load_conf_file('abrt.conf')
+    if settings.get('AutoreportingEnabled', 'no') in ['disabled', 'no', '0', 'off']:
```
Comment (mkutlak) [inline src/cli/abrtcli/cli.py]: One more thing: 
https://github.com/abrt/abrt/blob/d10caaf40e613b566430c2b015d604b4da9e65d1/abrt.spec.in#L522

https://github.com/abrt/abrt/blob/d10caaf40e613b566430c2b015d604b4da9e65d1/configure.ac#L279

Comment (ernestask) [reply src/cli/abrtcli/cli.py]: Removed them. The code can just be patched out on RHEL.

File: abrt.spec.in [hunk 7]
```diff
@@ -339,28 +339,21 @@ Requires: %{name} = %{version}-%{release}
...
 Requires: python3-abrt
 Requires: abrt-addon-ccpp
 Requires: python3-argh
 Requires: python3-argcomplete
 Requires: python3-humanize
 
-%description cli-ng
-New generation command line interface for ABRT
+Provides: %{name}-cli-ng = %{version}-%{release}
+Obsoletes: %{name}-cli-ng <= 2.12.1
```
Comment (mkutlak) [inline abrt.spec.in]:
```suggestion
Obsoletes: %{name}-cli-ng < 2.12.2
```

**Pull request merged**